apps/internal/base/base.go

```diff
@@ -309,6 +310,18 @@ func (b Client) Account(homeAccountID string) shared.Account {
 	return account
 }
 
+func (b Client) RemoveAccount(account shared.Account) error {
+	if s, ok := b.manager.(cache.Serializer); ok {
```

Comment:
I think its a good suggestion, I am unclear about some things though. 

We want the suggestedCacheKey() calculation at every acquireToken() call, so setSerializer() should not set it for once at the application level. 

Another thing is the cache interface is public, Serializer has marshal and unmarshal which if implemented by developers for their own cache should not get affected because of this change

I think I am a little unsure of what will move in manager, we can have a call to discuss this and if needed lets open up an issue to track this and make the changes in a separate PR as you said.